python/arcticdb/diagnostics_engine/library_options.py

```diff
@@ -0,0 +1,286 @@
+    is enabled, Arctic will resort to "pickling" the data when it encounters a type that it can't store in its native format.
+    
+    \n- Pickling is a process in Python for serializing and de-serializing a Python object structure. Any object
+    in Python can be pickled so that it can be saved on disk, and then later, the pickled file can be unpickled
+    back into the original object.
+    \n- However, pickling comes with certain limitations. One key drawback is that pickled data cannot be indexed.
+    This means you can't effectively perform operations like date range queries or column slicing, which are
+    common in data analysis tasks.
+    \n- Additionally, if a Pandas DataFrame contains even a single non-primitive data type object (i.e., not an
+    integer, float, string, or boolean), the entire DataFrame will need to be pickled. This could potentially
+    impact performance and the ease of data manipulation.
+    
+    \n For a more detailed understanding of pickling, including how it works and its various use cases and
+    limitations, please refer to the specific 'Pickling' section in this notebook.
+    """,
+    "symbol_list": """
+    \n If enabled, the symbol list caching functionality of Arctic will be used. This is designed to improve
+    performance of list_symbols operations. For a more detailed understanding of this feature, including how it
+    works and its various use cases and limitations, please refer to the specific 'Symbol list consistency
+    diagnosis' section in this notebook.""",
+}
+
+
+class LibraryInfoGetter:
+    """
+    This class provides a method for displaying and explaining the various options
+    associated with a specific library. It aims to provide users with a clear understanding
+    of each option, including its purpose, usage implications, and potential impact on
+    library operations.
+
+    Methods:
+    --------
+    display_library_options():
+        Fetches and displays the options for a given library, along with detailed
+        explanations for each option.
+
+    Example Usage:
+    --------------
+    info_getter = LibraryInfoGetter(lib, EXPLANATIONS)
+    info_getter.display_library_options()
+    """
+
+    def __init__(self, lib, explanations):
+        self.response_obj = lib._lib_cfg.lib_desc.version
+        self.explanations = explanations
+
+        self.response_non_default_str = (
+            "## Next, see the list of your library options that has been modified with respect to the default"
+            " values:\n\n"
+        )
+        self.response_default_str = (
+            "## Next, see the list of your library options that remains with the default values:\n\n"
+        )
+
+        # Default values defined internally
+        self.internal_default_values = {
+            "column_group_size": 127,
+            "segment_row_size": 100000,
+            "max_num_buckets": 150,
+        }
```

Comment:
I don't think this list is complete. If you create a new library within Man, then the protobuf defaults are overridden for a lot of settings.

Reply:
Apart from the boolean types, which don't allow the internal setting trick by comparison with zero, I've found only one additional item to add to this list. This is max_blob_size, which is set in the Python layer. I am surprised that you said "a lot". Am I missing something?